src/utils/logger/index.ts

```diff
@@ -0,0 +1,119 @@
+import { Logger, transports, LoggerInstance } from 'winston';
+import * as path from 'path';
+
+import { LoggerMethods, Loggers, LogMessage } from './types';
+import levels from './levels';
+
+let loggers: Loggers = null;
+
+/**
+ * Resolve path for filename in <root>/logs
+ * @param {string} filename the name of the log file
+ */
+function resolve(filename: string) {
+  return path.resolve('..', '..', '..', 'logs', filename);
+}
+
+/**
+ * Perform check to ensure that loggers are initialized before usage
+ */
+function check() {
+  if (!loggers) {
+    throw new Error('Logger was not initialized');
+  }
+}
+
+/**
+ * Generate a logger dynamically
+ *
+ * @param {string} name Name of logger to create
+ */
+function generateLogger(name: string) {
+  return {
+    [name]: new (Logger)({
+      transports: [
+        new (transports.File)({
+          name: `${name}-file`,
+          filename: resolve(`server-${name}.log`),
+          level: name,
+          json: true,
+          maxsize: 5242880,
+          maxFiles: 5,
+          handleExceptions: handleExceptions(name),
+          colorize: false,
+        }),
+
+        new (transports.Console)({
+          handleExceptions: handleExceptions(name),
+          colorize: true,
+        }),
+      ],
+    }),
+  };
+}
+
+function handleExceptions(name: string) {
+  switch (name) {
+    case 'error':
+      return true;
```

Comment:
Doesn't seem like much handling to me? What is this handling.

Reply:
Should rename this to shouldHandleExceptions. Also as I said in previous comment this logger is going to need to be heavily reworked to send the logs back to the server